src/adaptors/UBImportSMART.cpp

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (C) 2015-2018 Département de l'Instruction Publique (DIP-SEM)
+ *
+ * Copyright (C) 2013 Open Education Foundation
+ *
+ * Copyright (C) 2010-2013 Groupement d'Intérêt Public pour
+ * l'Education Numérique en Afrique (GIP ENA)
+ *
+ * This file is part of OpenBoard.
+ *
+ * OpenBoard is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 3 of the License,
+ * with a specific linking exception for the OpenSSL project's
+ * "OpenSSL" library (or with modified versions of it that use the
+ * same license as the "OpenSSL" library).
+ *
+ * OpenBoard is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with OpenBoard. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+
+#include <QDir>
+#include <QList>
+#include <QXmlStreamReader>
+
+#include "adaptors/UBSvgSubsetAdaptor.h"
+#include "core/UBApplication.h"
+#include "core/UBPersistenceManager.h"
+#include "core/UBDocumentManager.h"
+#include "core/UBPersistenceManager.h"
+#include "document/UBDocumentProxy.h"
+#include "frameworks/UBFileSystemUtils.h"
+
+#include "UBSvgSubsetAdaptor.h"
+
+#include "UBImportSMART.h"
+
+#include "globals/UBGlobals.h"
+
+THIRD_PARTY_WARNINGS_DISABLE
+#include "quazip.h"
+#include "quazipfile.h"
+#include "quazipfileinfo.h"
+THIRD_PARTY_WARNINGS_ENABLE
+
+#include "core/memcheck.h"
+
+UBImportSMART::UBImportSMART(QObject *parent)
+    : UBDocumentBasedImportAdaptor(parent)
+{
+    // NOOP
+}
+
+
+UBImportSMART::~UBImportSMART()
+{
+    // NOOP
+}
+
+
+QStringList UBImportSMART::supportedExtentions()
+{
+    QStringList formats = {"notebook", "xbk"};
+    return formats;
+}
+
+
+QString UBImportSMART::importFileFilter()
+{
+    QString filter = tr("SMART Notebook (");
+    QStringList formats = supportedExtentions();
+    bool isFirst = true;
+
+    foreach(QString format, formats)
```

Comment:
foreach is old Qt api, I'd suggest to make `formats´ const, and iterate through it using
```
for (const QString &format : formats) {
}
```